packages/player/src/components/player.tsx

```diff
@@ -0,0 +1,461 @@
+                  volume: (_context, event) => event.volume,
+                }),
+                (context, event) => {
+                  if (context.video) context.video.volume = event.volume
+                },
+              ],
+            },
+            PLAYBACKRATE_CHANGE: {
+              target: 'paused',
+              actions: [
+                assign({
+                  playbackRate: (_context, event) => event.playbackRate,
+                }),
+                (context, event) => {
+                  if (context.video)
+                    context.video.playbackRate = event.playbackRate
+                },
+              ],
+            },
+            ACTIVITY: {
+              target: 'paused',
+              actions: assign({
+                isActive: (_context, _event) => true,
+              }),
+            },
+            SEEKING: {
+              target: 'paused',
+              actions: [
+                assign({
+                  seekingTime: (context, event) => event.seekingTime,
+                }),
+                'seekVideo',
+              ],
+            },
+          },
+        },
+        playing: {
+          on: {
+            PAUSE: {target: 'paused', actions: ['pauseVideo']},
+            ACTIVITY: {
+              target: 'playing',
+              actions: assign({
+                isActive: (_context, _event) => true,
+              }),
+            },
+            VOLUME_CHANGE: {
+              target: 'playing',
+              actions: [
+                assign({
+                  volume: (_context, event) => event.volume,
+                }),
+                (context, event) => {
+                  if (context.video) context.video.volume = event.volume
+                },
+              ],
+            },
+            PLAYBACKRATE_CHANGE: {
+              target: 'playing',
+              actions: [
+                assign({
+                  playbackRate: (_context, event) => event.playbackRate,
+                }),
+                (context, event) => {
+                  if (context.video)
+                    context.video.playbackRate = event.playbackRate
+                },
+              ],
+            },
+            SEEKING: {
+              target: 'playing',
+              actions: [
+                assign({
+                  seekingTime: (context, event) => event.seekingTime,
+                }),
+                'seekVideo',
+              ],
+            },
+            END: 'ended',
+            TIMING: {
+              target: 'playing',
+              actions: assign({
+                currentTime: (context, _event) =>
+                  context.video?.currentTime ?? 0,
+                duration: (context, _event) => context.video?.duration ?? 0,
+              }),
+            },
+          },
+        },
+        ended: {on: {PLAY: 'playing'}},
+      },
+    },
+    failure: {},
+  },
+})
+
+interface VideoContextType {
+  videoService: ActorRefFrom<typeof videoMachine>
```

Comment:
the context provides an xstate service, which is basically an initialized and running machine that can be passed around the application